looper/conductor.py

```diff
@@ -62,6 +62,29 @@ def write_sample_yaml(namespaces):
     sample.to_yaml(get_sample_yaml_path(namespaces), add_prj_ref=False)
     return {"sample": sample}
 
+def write_namespaces(namespaces):
+    """
+    Save all namespaces to YAML.
+
+    :param dict namespaces: variable namespaces dict
+    :return dict: sample namespace dict
+    """
+    import yaml
+    # my_namespaces = namespaces["sample"].to_dict()
+    path = get_sample_yaml_path(namespaces)
+    my_namespaces = {}
+    for namespace, values in namespaces.items():
+        if namespace == "sample":
+            values.to_yaml(path)
+            my_namespaces.update({str(namespace):yaml.load(open(path))})
```

Comment:
why did you decide to read the sample.yaml in and update? There can be a case when the sample yaml file is not there.

Reply:
for the other namespaces, I can convert the object type to dict, then write to .yaml.  But for sample, I have't find a way to convert it to dict. so I first write the sample.yaml the load it as dict.

Reply:
That's not the optimal way to do this. Let's abstract the conversion of `Sample` to `dict` away from the `to_yaml` method and use it.

Reply:
> That's not the optimal way to do this. Let's abstract the conversion of `Sample` to `dict` away from the `to_yaml` method and use it.

do you want me to make the changes in `sample.py`? that's in `peppy` right?

Reply:
yes, this is in peppy, I'll do this.

Reply:
should be ready to use: https://github.com/pepkit/peppy/blob/e8ccfda73d8b5abab5b77ebae964d5ecb17ebcec/peppy/sample.py#L79-L126

Reply:
So this looper release will require an updated release to peppy?

Reply:
yes, as I noted in the main PR description

Reply:
not for this PR...